glue/viewers/profile/layer_artist.py

```diff
@@ -149,15 +149,17 @@ def _update_visual_attributes(self):
 
     def _update_profile(self, force=False, **kwargs):
 
-        if (self._viewer_state.x_att is None or
+        if (self._viewer_state.x_att_pixel is None or
                 self.state.attribute is None or
                 self.state.layer is None):
             return
 
         changed = set() if force else self.pop_changed_properties()
 
-        if force or any(prop in changed for prop in ('layer', 'x_att', 'attribute', 'function', 'normalize', 'v_min', 'v_max', 'visible')):
-            self._calculate_profile(reset=force)
+        if force or any(prop in changed for prop in ('layer', 'slices', 'x_att', 'x_att_pixel', 'attribute',
+                                                     'function', 'normalize', 'v_min', 'v_max', 'visible')):
+            self._update_visual_attributes()
+            self._calculate_profile(reset=True)
```

Comment:
For other viewers the approach is instead to do:

```
self._calculate_profile(reset=True)
force=True
```

which ensures the visual properties are updated next. Maybe switch to this approach?

Reply:
Sure

Reply:
Tried changing to 

```python
    self._calculate_profile(reset=True)
    self._update_profile(force=True)
```
But error was thrown upon attempting to view via 1D Profile viewer as data cannot be added for some reason...